app/v2/models/responses/hateoas/HateoasNewMovementResponse.scala

```diff
@@ -36,17 +41,14 @@ object HateoasNewMovementResponse extends HateoasResponse {
       "messages" -> Json.obj("href" -> getMessagesUri(movementId, None, movementType))
     )
 
-    Json.obj("_links" -> jsObject) ++
-      boxResponse
-        .map(
-          r => Json.obj("boxId" -> r.boxId.value)
-        )
-        .getOrElse(Json.obj()) ++
-      upscanInitiateResponse
-        .map(
-          r => Json.obj("uploadRequest" -> Json.obj("href" -> r.uploadRequest.href, "fields" -> r.uploadRequest.fields))
-        )
-        .getOrElse(Json.obj())
+    Json.obj("_links" -> jsObject) ++ {
+      (boxResponse, upscanInitiateResponse) match {
+        case (None, None)                       => Json.obj(getMovementId(movementType) -> movementId.value)
```

Comment:
This bit should be added regardless. It's currently only being added if there is no box response or upscan response.

Reply:
:+1: